components/FrontMatter.js

```diff
@@ -0,0 +1,24 @@
+import { Text } from "@asyncapi/generator-react-sdk";
+
+import * as fs from "fs";
+import * as yaml from "yaml";
+
+export function FrontMatter({ asyncapi, params }) {
+  if (!(params.slate || params.frontMatter)) return '';
+
+  let frontMatter;
+  if (params.frontMatter) {
+    frontMatter = yaml.parse(fs.readFileSync(params.frontMatter,'utf8'));
+  } else {
+    frontMatter = {
+      title: asyncapi.info().title(),
+      language_tabs: [],
+      toc_footers: [],
+      includes: [],
+      search: true,
+      code_clipboard: true
+    };
+  }
+  const frontMatterStr = yaml.stringify(frontMatter);
+  return <Text newLines={2}>{`---\n${frontMatterStr}\n---`}</Text>
```

Comment:
what we get in the markdown file is
```
---
title: My AsyncAPI
language_tabs:
  - shell: Shell
  - javascript: Node.js
toc_footers:
  - Built with AsyncAPI/Generator
includes: []
search: true
code_clipboard: true

---
```

is this extra space, empty line ok for markdown ? won't it cause any problems?

Reply:
Yes, I think blank lines between keys are fine in YAML, we could tweak it though.

Reply:
that is fine, if it works well this way there is no need to change it

Reply:
Extraneous `\n` removed.